ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/workshopAdmin/presenter/WorkshopAdminPresenterImpl.java

```diff
@@ -0,0 +1,178 @@
+import org.iplantc.de.client.models.collaborators.Collaborator;
+import org.iplantc.de.client.models.groups.GroupAutoBeanFactory;
+import org.iplantc.de.client.models.groups.Member;
+import org.iplantc.de.client.models.groups.MemberSaveResult;
+import org.iplantc.de.collaborators.client.events.UserSearchResultSelected;
+import org.iplantc.de.commons.client.ErrorHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author dennis
+ */
+public class WorkshopAdminPresenterImpl implements WorkshopAdminView.Presenter {
+
+    private final WorkshopAdminView view;
+    private final WorkshopAdminServiceFacade serviceFacade;
+    private final GroupAutoBeanFactory groupAutoBeanFactory;
+    private final WorkshopAdminView.WorkshopAdminViewAppearance appearance;
+    private final ListStore<Member> listStore;
+
+    private final class UserSearchResultSelectedEventHandler
+            implements UserSearchResultSelected.UserSearchResultSelectedEventHandler {
+
+        private Member memberFromCollaborator(Collaborator collaborator) {
+            Member member = groupAutoBeanFactory.getMember().as();
+            member.setId(collaborator.getUserName());
+            member.setAttributes(new ArrayList<String>());
+            member.setEmail(collaborator.getEmail());
+            member.setFirstName(collaborator.getFirstName());
+            member.setInstitution(collaborator.getInstitution());
+            member.setLastName(collaborator.getLastName());
+            member.setName(collaborator.getName());
+            return member;
+        }
+
+        private boolean listContainsMember(List<Member> members, Member newMember) {
+            String newMemberId = newMember.getId();
+            if (newMemberId != null) {
+                for (Member member : members) {
+                    if (member.getId() != null && member.getId().equals(newMemberId)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        @Override
+        public void onUserSearchResultSelected(UserSearchResultSelected event) {
+
+            // Ignore the event if it wasn't initiated by the workshop admin panel.
+            if (!event.matchesTag(view.userSearchEventTag)) {
+                return;
+            }
+
+            // Add the user to the list if not there already.
+            Member member = memberFromCollaborator(event.getCollaborator());
+            if (!listContainsMember(listStore.getAll(), member)) {
+                listStore.add(member);
+            }
+        }
+    }
+
+    private final class DeleteMembersClickedEventHandler
+            implements DeleteMembersClickedEvent.DeleteMembersClickedEventHandler {
+
+        @Override
+        public void onDeleteMembersClicked(DeleteMembersClickedEvent event) {
+            for (Member member : event.getMembers()) {
+                listStore.remove(member);
```

Comment:
We normally don't have multiple simultaneous workshops, but there's no reason that we couldn't. The purpose of this group is to direct jobs submitted by users in workshops to a dedicated pool of Condor execute nodes, so we can accommodate simultaneous workshops by adding the users in all of the workshops to the workshop group.

I chose not to call the service when users are added to or removed from the list store because the service call is fairly slow. No changes are made until the user clicks the Save button. Hopefully, that won't be too confusing for workshop administrators.